test/controllers/predicates/InflightPPOBPredicateSpec.scala

```diff
@@ -0,0 +1,166 @@
+    when(mockVatSubscriptionService.getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext]))
+      .thenReturn(Future.successful(result))
+  }
+
+  def inflightPredicate: InflightPPOBPredicate =
+    new InflightPPOBPredicate(
+      mockVatSubscriptionService,
+      mockEnrolmentsAuthService,
+      mockErrorHandler,
+      messagesApi,
+      mockConfig
+    )
+
+  def target(result: GetCustomerInfoResponse = Right(customerInfoPendingAddressModel)): Action[AnyContent] = {
+    setup(result)
+    mockIndividualAuthorised()
+    inflightPredicate.async {
+      implicit user => Future.successful(Ok("test"))
+    }
+  }
+
+  "The InflightPPOBPredicateSpec" when {
+
+    "there is an inflight indicator in session" when {
+
+      "the inflight indicator is set to 'true'" should {
+
+        lazy val result = await(target()(request.withSession(inflightPPOBKey -> "true")))
+        lazy val document = Jsoup.parse(bodyOf(result))
+
+        "return 200" in {
+          status(result) shouldBe Status.OK
+        }
+
+        "show the 'PPOB change pending' error page" in {
+          document.title shouldBe "We are reviewing your request"
+        }
+
+        "not call the VatSubscriptionService" in {
+          verify(mockVatSubscriptionService, never())
+            .getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext])
+        }
+      }
+
+      "the inflight indicator is set to 'false'" should {
+
+        lazy val result = await(target()(request.withSession(inflightPPOBKey -> "false")))
+
+        "return 200" in {
+          status(result) shouldBe Status.OK
+        }
+
+        "not call the VatSubscriptionService" in {
+          verify(mockVatSubscriptionService, never())
+            .getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext])
+        }
+      }
+
+      "the inflight indicator is set to 'error'" should {
+
+        lazy val result = await(target()(request.withSession(inflightPPOBKey -> "error")))
+
+        "return 500" in {
+          status(result) shouldBe Status.INTERNAL_SERVER_ERROR
+        }
+
+        "not call the VatSubscriptionService" in {
+          verify(mockVatSubscriptionService, never())
+            .getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext])
+        }
+      }
+    }
+
+    "there is no inflight indicator in session" when {
+
+      "the user has an inflight PPOB address" should {
+
+        lazy val result = await(target()(request))
+        lazy val document = Jsoup.parse(bodyOf(result))
+
+        "return 200" in {
+          status(result) shouldBe Status.OK
+        }
+
+        "add the inflight indicator 'true' to session" in {
+          session(result).get(inflightPPOBKey) shouldBe Some("true")
+        }
+
+        "show the 'PPOB change pending' error page" in {
+          document.title shouldBe "We are reviewing your request"
+        }
+      }
+
+      "the user has an inflight email" should {
```

Comment:
Can you add a test for the view? Just to prove it loads a different view than the PPOB change pending page